l10n_it_sct_cbi/tests/test_generate_file.py

```diff
@@ -0,0 +1,170 @@
+
+        cls.bank_journal = cls.company_data["default_journal_bank"]
+        cls.bank_journal.bank_account_id = cls.company_bank_account
+
+        cls.supplier_bank = cls.env["res.bank"].create(
+            {
+                "name": "Test supplier bank",
+                "bic": "TESTBICA",
+            }
+        )
+
+        cls.supplier = cls.env["res.partner"].create(
+            {
+                "name": "Test supplier",
+                "bank_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "acc_number": "IT48N0300203280543765183341",
+                            "bank_id": cls.supplier_bank.id,
+                        },
+                    ),
+                ],
+            }
+        )
+
+        cls.payment_method = cls.env.ref("l10n_it_sct_cbi.sepa_cbi_credit_transfer")
+
+        payment_mode_form = Form(cls.env["account.payment.mode"])
+        payment_mode_form.name = "Test SEPA CBI payment mode"
+        payment_mode_form.bank_account_link = "fixed"
+        payment_mode_form.fixed_journal_id = cls.bank_journal
+        payment_mode_form.payment_method_id = cls.payment_method
+        cls.payment_mode = payment_mode_form.save()
+
+    def _get_record_from_action(self, action):
+        record_id = action["res_id"]
+        record_model = action["res_model"]
+        return self.env[record_model].browse(record_id)
+
+    def _get_payment_order(self, invoices):
+        payment_order_action = invoices.create_account_payment_line()
+        return self._get_record_from_action(payment_order_action)
+
+    def _get_payment_attachment(self, invoices):
+        payment_order = self._get_payment_order(invoices)
+        payment_order.draft2open()
+        payment_file_action = payment_order.open2generated()
+        return self._get_record_from_action(payment_file_action)
+
+    def test_2_bills(self):
+        """Generate a payment file for 2 vendor bills."""
+        # Arrange
+        payment_mode = self.payment_mode
+        supplier = self.supplier
+
+        bill_1 = self.init_invoice(
+            "in_invoice",
+            partner=supplier,
+            amounts=[
+                100,
+            ],
+            post=True,
+        )
+        bill_1.ref = "Test invoice 1"
+        bill_2 = self.init_invoice(
+            "in_invoice",
+            partner=supplier,
+            amounts=[
+                200,
+            ],
+            post=True,
+        )
+        bill_2.ref = "Test invoice 2"
+        bills = bill_1 | bill_2
+        bills.payment_mode_id = payment_mode
+
+        # Act
+        payment_att = self._get_payment_attachment(bills)
+
+        # Assert
+        payment_tree = etree.fromstring(base64.b64decode(payment_att.datas))
+        namespaces = payment_tree.nsmap
+        if None in namespaces:
+            namespaces["default"] = namespaces.pop(None)
```

Comment:
**question**: Non sono sicuro di questa aggiunta (e l'analoga qualche riga sotto)

@SirAionTech cosa ne pensi?

Reply:
Non so se un `find` con questi nuovi `namespaces` possa trovare i tag del namespace _incriminato_ (`namespaces[None]`), però è anche vero che per ora vengono solo cercati i tag nel namespace `PMRQ` quindi il test per ora credo sia rimasto valido

Reply:
Confermo che in locale i test passano anche senza questa modifica. Se non da problemi sulle pipeline OCA, sarei per rimuoverlo.

Per info, nel test `test_multiple_payment_priority` la variabile `namespaces` contiene questo:

```py
{
    None: 'urn:CBI:xsd:CBIBdyPaymentRequest.00.04.01', 
    'PMRQ': 'urn:CBI:xsd:CBIPaymentRequest.00.04.01'
}
```

Reply:
Confermo che anche nel mio ambiante doodba non ci sono errori di nessuna tipo ma su OCA si. Mi sono confrontato un po anche con @SirAionTech su Discord e dovrebbe essere il fatto che usa una versione di lxml diversa. Su oca è la 3.7.1 https://github.com/OCA/l10n-italy/actions/runs/11435980402/job/31812452602?pr=4367#step:4:228 mentre su doodba e sulla 16.0 è la 4.6.5

Con il debugger ho capito che il name space che viene fornito a find rompe la xpath che non può prendere argomenti nulli a quanto pare sulla versione 4.6.5 questo non è un problema.
Da questo ne è uscita la mia "rinomina" della voce None.